api/src/main/java/org/openmrs/ui/framework/UiUtils.java

```diff
@@ -431,6 +431,24 @@ public String escapeJs(String input) {
 		input = input.replaceAll("\"", "\\\\\"");
 		return input;
 	}
+
+	/**
+	 * Used to make sure that a string can safely be used in a URL for https transfer
+	 * 
+	 * @param input the input text which may have characters that an URL requires to escape
+	 * @return text that is fit to be used in an URL
+	 */
+	public String encodeForSafeURL(String input) {
+		if (input == null) {
+			return null;
+		}

```

Comment:
Isn't this method already provided by publicly available libraries?

Reply:
Yes @dkayiwa , the method does exist and refactoring this would then look like

```
public String encodeForSafeURL(String input) {
           return org.owasp.encoder.Encode.forUriComponent(input);
}
```

However the down side to using this and any other available library (e.g `java.net.URLEncoder`) would be too long and urgly a URL since nearly all the non alphanumeric characters are encoded for a need that can simply solved and that json is not likely to evolve to that level. This is just my thought but stand to be directed. Is this fine to go ahead and implement?

Thank you.

Reply:
What is the problem with long and ugly urls in comparison to the code we have to maintain ourselves?

Reply:
Good lesson for me here, thank you @dkayiwa . Let me refactor this implementation accordingly.

Reply:
Feel free to put forth your arguments if you feel otherwise. You know there is a lot i can also learn from you. You do not have to just take whatever i say. 😊